sql/modules/Menu.sql

```diff
@@ -102,6 +112,18 @@ It is thus similar to menu_generate() but it only returns the menu items
 associated with nodes directly descendant from the parent.  It is used for
 menues for frameless browsers.$$;
 
+CREATE OR REPLACE FUNCTION menu_item(in_id int) RETURNS SETOF menu_item
+AS $$
+SELECT * FROM menu_generate() where id = $1;
+$$ language sql;
```

Comment:
Here we already have the entire tree, just don't pa it to our callers. How much impact would it be to send it?